test/python/primitives/test_backend_estimator_dev.py

```diff
@@ -0,0 +1,924 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for BackendEstimator."""
+
+from __future__ import annotations
+
+from collections.abc import Iterator, Sequence
+from unittest import TestCase
+from unittest.mock import MagicMock, Mock, patch
+from test import combine
+
+from ddt import ddt, data, unpack
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.primitives import EstimatorResult
+from qiskit.primitives.backend_estimator_dev import (
+    AbelianDecomposer,
+    BackendEstimator,
+    NaiveDecomposer,
+)
+from qiskit.providers import Backend, Options, JobV1
+from qiskit.providers.fake_provider import FakeNairobi, FakeNairobiV2
+from qiskit.quantum_info.operators import SparsePauliOp
+from qiskit.quantum_info.operators.symplectic.pauli import Pauli
+from qiskit.quantum_info.operators.symplectic.pauli_list import PauliList
+from qiskit.result import Counts, Result
+from qiskit.transpiler import Layout, PassManager
+from qiskit.transpiler.passes import ApplyLayout, SetLayout
+from qiskit.test import QiskitTestCase
+
+BACKENDS = [FakeNairobi(), FakeNairobiV2()]
+
+
+################################################################################
+## AUXILIARY
+################################################################################
```

Comment:
These big comments (certainly imo) just use up screen space unnecessarily; they don't add anything, and they're liable to get out of date when somebody later updates the file.

Reply:
They have been very helpful to me developing, to organize families of functions together and traverse the workflow in a systematic way. I have left them there because I think they may help others further down the road (there quite a few functions), but if you really want them gone it is an easy fix.

Reply:
I agree with Jake. These big comments look unnecessary to me too. They could be debt of the code when someone else updates the code in the future. Because there is no automatic check of the integrity of these big comments, contributors and reviewers may miss it.

Reply:
I have removed a bunch of these comments, I'll keep working towards getting rid of most of them